datasets/medmcqa/medmcqa.py

```diff
@@ -0,0 +1,131 @@
+}
+"""
+
+
+class MedMCQA(datasets.GeneratorBasedBuilder):
+    """MedMCQA : A Large-scale Multi-Subject Multi-Choice Dataset for Medical domain Question Answering"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "question": datasets.Value("string"),
+                "opa": datasets.Value("string"),
+                "opb": datasets.Value("string"),
+                "opc": datasets.Value("string"),
+                "opd": datasets.Value("string"),
+                "cop": datasets.features.ClassLabel(names=["a", "b", "c", "d"]),
+                "choice_type": datasets.Value("string"),
+                "exp": datasets.Value("string"),
+                "subject_name": datasets.Value("string"),
+                "topic_name": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train.json"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "test.json"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "dev.json"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+
+        with open(filepath, encoding="utf-8") as f:
+            for key, row in enumerate(f):
+                data = json.loads(row)
+                yield key, {
+                    "id": data["id"],
+                    "question": data["question"],
+                    "opa": data["opa"],
+                    "opb": data["opb"],
+                    "opc": data["opc"],
+                    "opd": data["opd"],
+                    "cop": -1 if split == "test" else int(data["cop"]) - 1,
+                    "choice_type": data["choice_type"],
+                    "exp": "" if split == "test" else data["exp"],
+                    "subject_name": data["subject_name"],
+                    "topic_name": data["topic_name"],
+                }
```

Comment:
I think this can be simplified: no need to pass "split"
```suggestion
    def _split_generators(self, dl_manager):
        """Returns SplitGenerators."""
        data_dir = dl_manager.download_and_extract(_URL)
        return [
            datasets.SplitGenerator(
                name=datasets.Split.TRAIN,
                gen_kwargs={
                    "filepath": os.path.join(data_dir, "train.json"),
                },
            ),
            datasets.SplitGenerator(
                name=datasets.Split.TEST,
                gen_kwargs={
                    "filepath": os.path.join(data_dir, "test.json"),
                },
            ),
            datasets.SplitGenerator(
                name=datasets.Split.VALIDATION,
                gen_kwargs={
                    "filepath": os.path.join(data_dir, "dev.json"),
                },
            ),
        ]

    def _generate_examples(self, filepath):
        with open(filepath, encoding="utf-8") as f:
            for key, row in enumerate(f):
                data = json.loads(row)
                data["cop"] = int(data.get("cop", 0)) - 1
                data["exp"] = data.get("exp", "")
                yield key, data
```

Reply:
Thanks for the suggestion.